src/core/CMakeLists.txt

```diff
@@ -181,11 +183,15 @@ endif(SCAFACOS)
 
 # Subdirectories
 add_subdirectory(io)
-
+if(H5MD)
+target_link_libraries(EspressoCore PUBLIC ${HDF5_LIBRARIES} Boost::filesystem h5xx)
+target_sources(EspressoCore PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/io/writer/h5md_core.cpp)
+target_include_directories(EspressoCore SYSTEM PUBLIC ${HDF5_INCLUDE_DIRS} ${CMAKE_CURRRENT_SOURCE_DIR}/io/writer)
+target_compile_definitions(EspressoCore PUBLIC H5XX_USE_MPI)
+endif(H5MD)
```

Comment:
This change actually fixes a bug in ESPResSo 4.1.4. In particular, the `${HDF5_LIBRARIES}` needs to be linked against `EspressoCore`, otherwise it is not possible to link `ScriptInterface` unit tests in recent versions of openSUSE due to missing hdf5 symbols. For example, on openSUSE Leap 15.4 x86_64:
```
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld:
    /home/abuild/rpmbuild/BUILD/espresso/build/src/core/io/writer/H5mdCore.so:
    undefined reference to `H5Pset_fapl_mpio'
collect2: error: ld returned 1 exit status
make[3]: *** [src/core/unit_tests/CMakeFiles/AutoParameters_test.dir/build.make:112:
              src/core/unit_tests/AutoParameters_test] Error 1
```